iidm/iidm-api/src/main/java/com/powsybl/iidm/network/Line.java

```diff
@@ -22,4 +24,106 @@
 
     boolean isTieLine();
 
+    /**
+     * Move the line's end on side ONE to the given node of the given voltage level.
+     * If the given voltage level's topology is not NODE-BREAKER, a runtime exception is thrown.
+     * If the topology of the voltage levels previously at the ends of the line is not NODE-BREAKER,
+     * a runtime exception is also thrown.
+     */
+    default Line move1(int node, VoltageLevel voltageLevel) {
+        return move(node, voltageLevel, getTerminal2().getNodeBreakerView().getNode(), getTerminal2().getVoltageLevel());
+    }
+
+    /**
+     * Move the line's end on side TWO to the given node of the given voltage level.
+     * If the given voltage level's topology is not NODE-BREAKER, a runtime exception is thrown.
+     * If the topology of the voltage levels previously at the ends of the line is not NODE-BREAKER,
+     * a runtime exception is also thrown.
+     */
+    default Line move2(int node, VoltageLevel voltageLevel) {
+        return move(getTerminal1().getNodeBreakerView().getNode(), getTerminal1().getVoltageLevel(), node, voltageLevel);
+    }
+
+    /**
+     * Move the line's ends to the given nodes of the given voltage levels.
+     * If the given voltage levels' topology is not NODE-BREAKER, a runtime exception is thrown.
+     * If the topology of the voltage levels previously at the ends of the line is not NODE-BREAKER,
+     * a runtime exception is also thrown.
+     */
+    default Line move(int node1, VoltageLevel voltageLevel1, int node2, VoltageLevel voltageLevel2) {
+        if (voltageLevel1.getTopologyKind() != TopologyKind.NODE_BREAKER
+                || voltageLevel2.getTopologyKind() != TopologyKind.NODE_BREAKER
+                || getTerminal1().getVoltageLevel().getTopologyKind() != TopologyKind.NODE_BREAKER) {
```

Comment:
A branch is not allowed to be between two voltage levels with a different topology so that seems a bit redundant? But it is not documented... It should be.